src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -0,0 +1,980 @@
+    size_t i       = 0;
+
+    while (newDstL.Next())
+    {
+        dstList[i] = newDstL.GetValue();
+        i++;
+    }
+
+    if (CHIP_NO_ERROR != newDstL.GetStatus())
+    {
+        mTimeSyncDataProvider.LoadDSTOffset(mDstOffsetObj);
+        return CHIP_IM_GLOBAL_STATUS(InvalidCommand);
+    }
+
+    mDstOffsetObj.size = i;
+
+    // only 1 validuntil null value and shall be last in the list
+    uint64_t lastValidUntil = 0;
+    for (i = 0; i < mDstOffsetObj.size; i++)
+    {
+        const auto & dstItem = GetDSTOffset()[i];
+        // list should be sorted by validStarting
+        // validUntil shall be larger than validStarting
+        if (!dstItem.validUntil.IsNull() && dstItem.validStarting >= dstItem.validUntil.Value())
+        {
+            mTimeSyncDataProvider.LoadDSTOffset(mDstOffsetObj);
+            return CHIP_ERROR_INVALID_TIME;
+        }
+        // validStarting shall not be smaller than validUntil of previous entry
+        if (dstItem.validStarting < lastValidUntil)
+        {
+            mTimeSyncDataProvider.LoadDSTOffset(mDstOffsetObj);
+            return CHIP_ERROR_INVALID_TIME;
+        }
+        lastValidUntil = !dstItem.validUntil.IsNull() ? dstItem.validUntil.Value() : lastValidUntil;
+        // only 1 validUntil null value and shall be last in the list
+        if (dstItem.validUntil.IsNull() && (i != mDstOffsetObj.size - 1))
+        {
+            mTimeSyncDataProvider.LoadDSTOffset(mDstOffsetObj);
+            return CHIP_ERROR_INVALID_TIME;
+        }
+    }
+
+    return mTimeSyncDataProvider.StoreDSTOffset(GetDSTOffset());
+}
+
+CHIP_ERROR TimeSynchronizationServer::ClearTimeZone()
+{
+    mTimeZoneObj.size         = 1; // one default time zone item is needed
+    mTimeZoneObj.timeZoneList = Span<TimeZoneStore>(mTz);
+    for (auto & tzObj : mTimeZoneObj.timeZoneList)
+    {
+        memset(tzObj.name, 0, sizeof(tzObj.name));
+        tzObj.timeZone = { .offset = 0, .validAt = 0, .name = MakeOptional(CharSpan(tzObj.name, sizeof(tzObj.name))) };
+    }
+    return mTimeSyncDataProvider.StoreTimeZone(GetTimeZone());
+}
+
+CHIP_ERROR TimeSynchronizationServer::ClearDSTOffset()
+{
+    mDstOffsetObj.size = 0;
+    return mTimeSyncDataProvider.ClearDSTOffset();
+}
+
+DataModel::Nullable<Structs::TrustedTimeSourceStruct::Type> & TimeSynchronizationServer::GetTrustedTimeSource()
+{
+    return mTrustedTimeSource;
+}
+
+CHIP_ERROR TimeSynchronizationServer::GetDefaultNtp(MutableCharSpan & dntp)
+{
+    return mTimeSyncDataProvider.LoadDefaultNtp(dntp);
+}
+
+Span<TimeZoneStore> TimeSynchronizationServer::GetTimeZone()
+{
+    return mTimeZoneObj.timeZoneList.SubSpan(0, mTimeZoneObj.size);
+}
+
+DataModel::List<Structs::DSTOffsetStruct::Type> TimeSynchronizationServer::GetDSTOffset()
+{
+    return mDstOffsetObj.dstOffsetList.SubSpan(0, mDstOffsetObj.size);
+}
+
+void TimeSynchronizationServer::ScheduleDelayedAction(System::Clock::Seconds32 delay, System::TimerCompleteCallback action,
```

Comment:
This seems to be unused.  Followup to either start using it or remove it.

Reply:
Tracked in https://github.com/project-chip/connectedhomeip/issues/27575, resolving this comment for now, further followup on the issue.